onnxruntime/core/optimizer/attention_fusion.cc

```diff
@@ -111,7 +111,7 @@ static NodeArg& MergeQkvWeights(Graph& graph, int64_t hidden_size,
     utils::SetRawDataInTensorProto(initializer, result.data(), gsl::narrow<size_t>(element_count) * sizeof(MLFloat16));
   }
 
-  return graph_utils::AddInitializer(graph, initializer);
+  return graph_utils::AddInitializerWithExternalData(graph, initializer);
```

Comment:
what ensures that the data is external given `graph_utils::AddInitializerWithExternalData` has

`  ORT_ENFORCE(!has_external_data, "Expecting an initializer that contains data inline");`